File: src/ships/deriveFuelCapacity.ts
import type { TierColor } from "../data/types/tierColor.ts";
import { FUEL_CAPACITY_BY_TIER } from "../data/constants/shipsAndTravelConfig.ts";

// Ship Fuel (profitable-design-questions.md). Same shape as
// deriveShipTier.ts's tierMidpoint() -- a pure tier-keyed lookup, never
// reimplemented inline anywhere fuel capacity is needed.
export function deriveFuelCapacity(tier: TierColor): number {
  const entry = FUEL_CAPACITY_BY_TIER.find((e) => e.tier === tier);
  if (!entry) throw new RangeError(`no fuel capacity defined for tier ${tier}`);
  return entry.capacity;
}

File: unity/ProfitableCore/Simulation/DeriveFuelCapacity.cs
using Profitable.Core.Constants;
using Profitable.Core.Schema;

namespace Profitable.Core.Simulation;

// Ports src/ships/deriveFuelCapacity.ts. A pure tier-keyed lookup, never
// reimplemented inline anywhere fuel capacity is needed.
public static class FuelCapacityDeriver
{
    public static double DeriveFuelCapacity(TierColor tier)
    {
        if (!ShipsAndTravelConfig.FuelCapacityByTier.TryGetValue(tier, out var capacity))
        {
            throw new InvalidOperationException($"no fuel capacity defined for tier {tier}");
        }
        return capacity;
    }
}
